### PR TITLE
fix: 2234 - Remove SafeAreaView in favor of useSafeAreaInsets hook

### LINKED ISSUE #2234: [Maverick] Demo screens glitch on first render
### Describe the bug

When switching between demo screens using the bottom tab navigator, the first render of each screen glitches briefly – the content is rendered too high then jumps down to the correct position after a frame or two.  Subsequent renders of the screens are fine.

It's not noticeable on the simulator but is on both Android and iOS devices – below is a screen recording from a Pixel 6.
- On Android devices, the issue goes away if I remove the `<StatusBar>` item from the `Screen` component (but then the status bar doesn't pick up the app theme).
- On iOS devices, the `<StatusBar>` item isn't causing the problem – it looks like it's the `<SafeAreaView>` instead.
- I thought that adding `lazy: false` to the `<Tab.Navigator screenOptions` might be a reasonable workaround, but it doesn't seem to make a difference.

https://user-images.githubusercontent.com/17591275/193025794-c5c9090d-16fd-43f2-b4e9-188f0690e1c3.mp4

[Edit]: Here's a slowed-down video to show the glitches more clearly

https://user-images.githubusercontent.com/17591275/193048629-18140301-93d8-472a-9769-b4528a93fb97.mp4

### Ignite version

8.1.0

### Additional info

```
npx ignite-cli doctor

System
  platform           darwin                                           
  arch               arm64                                            
  cpu                10 cores         Apple M1 Max                    
  directory          IgniteMaverick   /Users/rb/GitHub/IgniteMaverick 

JavaScript (and globally-installed packages)
  node               18.4.0       /Users/rb/.nvm/versions/node/v18.4.0/bin/node 
  npm                8.12.1       /opt/homebrew/bin/npm                         
    expo-cli         5.4.12                                                     
    npm              8.12.1                                                     
  yarn               1.22.19      /opt/homebrew/bin/yarn                        
    react-devtools   4.24.7                                                     
  pnpm               -            not installed                                 
  expo               46.0.10      managed                                       

Ignite
  ignite-cli         8.1.0        /Users/rb/.npm/_npx/e31027f3785124a8/node_modules/.bin/ignite      
  ignite src         build        /Users/rb/.npm/_npx/e31027f3785124a8/node_modules/ignite-cli/build 

Android
  java               11.0.16      /Library/Java/JavaVirtualMachines/zulu-11.jdk/Contents/Home/bin/java 
  android home       -            /Users/rb/Library/Android/sdk                                        

iOS
  xcode              14.0.1     
  cocoapods          1.11.3       /opt/homebrew/bin/pod 

Tools
  git                git version 2.37.2   /opt/homebrew/bin/git 
```

### PR DESCRIPTION
**Pull request by carlinisaacson**
## Please verify the following:

- [x] `yarn test` **jest** tests pass with new tests, if relevant
- [ ] ~`README.md` has been updated with your changes, if relevant~

## Describe your PR

There is an issue with the SafeAreaView from react-native-safe-area-context that causes the top inset to not be set on first render. So you see the content, then it re-renders and jumps down to it's final resting place. The fix is to not use SafeAreaView and use the useSafeAreaInsets hook instead.

This PR changes the Screen component to use the hook and apply the insets as screen padding if they are passed in, which fixes #2234.

### Old behavior

https://user-images.githubusercontent.com/1761434/193369617-27e9df6b-5489-4ee2-bc25-2026ae9e1414.MP4

### New (fixed) behavior

https://user-images.githubusercontent.com/1761434/193369658-838e3a9b-2469-43c5-b4c2-f47e6d5ae0f3.MP4

### Multiple edges passed in (top and bottom)

Ugly, but it shows it still works

top-and-bottom

### CODE REVIEW

**Review by frankcalise — APPROVED** [review]
Tested it on my device and definitely fixed the jank for me! Nicely done.

LGTM

**Review by mazenchami — APPROVED** [review]
This looks great 🚀

File: boilerplate/app/components/Screen.tsx [hunk 1]
```diff
@@ -197,18 +205,15 @@ export function Screen(props: ScreenProps) {
...
-  const Wrapper = safeAreaEdges?.length ? SafeAreaView : View
+  const insetPadding = useSafeAreaInsetPadding(safeAreaEdges)
 
   return (
-    <Wrapper
-      edges={safeAreaEdges}
-      {...SafeAreaViewProps}
-      style={[$safeAreaStyle, SafeAreaViewProps?.style, { backgroundColor }]}
+    <View
+      style={[$containerStyle, { backgroundColor, ...insetPadding }]}
```
Comment (jamonholmgren) [inline boilerplate/app/components/Screen.tsx:216]: Probably prefer this style.

```suggestion
      style={[$containerStyle, { backgroundColor }, insetPadding ]}
```

**Review by jamonholmgren — APPROVED** [review]

**Pull request merged**